python/in_toto_attestation/v1/resource_descriptor.py

```diff
@@ -0,0 +1,37 @@
+#  Wrapper class for in-toto attestation ResourceDescriptor protos.
+
+import in_toto_attestation.v1.resource_descriptor_pb2 as rdpb
+from google.protobuf.struct_pb2 import Value
+import google.protobuf.json_format as pb_json
+import json
+
+class ResourceDescriptor:
+    def __init__(self, name='', uri='', digest=None, content=bytes(), download_location='', media_type='', annotations=None) -> None:
+        self.pb = rdpb.ResourceDescriptor()
+        self.pb.name = name
+        self.pb.uri = uri
+        if digest:
+            self.pb.digest.update(digest)
+        self.pb.content = content
+        self.pb.downloadLocation = download_location
+        self.pb.mediaType = media_type
+
+        # HACK
+        if annotations:
+            # Value type fields are considered Messages in Python.
+            # this is needed to set Value fields into the annotations map
+            # without having to know its value type a priori.
+            for k, v in annotations.items():
+                val_pb = pb_json.Parse(json.dumps(v), Value())
+                self.pb.annotations[k].CopyFrom(val_pb)
```

Comment:
If any one has a suggestion for cleaning this up, I'm all ears. The switch to using Values in `annotations` since #251  actually complicated the Python API for ResourceDescriptors, as Value is an actual protobuf message (i.e., an arbitrary value can't just be assigned via dicts as was the case when it was a Struct/object). 

After a fair amount of searching, using json marshalling seems like the most succinct way of setting arbitrary Values in the `annotations` map. The alternative, as far as I found, is to set the individual `int_value/string_value/etc` properties in the Value proto, which would require us to check the type of each `annotations` field to set the appropriate property.

Reply:
cc @joshuagl @adityasaky

Reply:
I didn't think about it before in https://github.com/in-toto/attestation/pull/251 and the corresponding issue but should we quickly amend it again to just use a Struct instead of map[string]Value? Struct is essentially a map[string]Value anyway, but it's easier to handle from a dict and such.

If we don't want to tweak it again, I think this hack is probably the best option.

Reply:
I'm not sure why this is considered a hack?

I might avoid the additional nesting under the `if` by doing something like:
```Python
annotations = annotations or {}
# Value type fields are considered Messages in Python.
# this is needed to set Value fields into the annotations map
# without having to know its value type a priori.
for k, v in annotations.items():
    # ...
```

Reply:
I don't think we should revert the ResourceDescriptor, though I do feel we should have thought a little more carefully about the implications of the use of Value in the `annotations` field. 

I think the current solution feels very much like a hack because its a lazy workaround for the fact that the protobuf object cannot handle the conversion of arbitrary Python values to the appropriate Value internally. The Struct type we had in `annotations` earlier does do this, so either I'm missing something or this isn't how protobuf expects Value fields to be used.

Reply:
So we had `map[string]Struct`, and updated it to `map[string]Value` which is essentially the structure of `Struct`. Had we updated to using `Struct` for annotations instead of `map[string]Value`, I think we'd get the `Value` handling from the library.

Reply:
Ah, I misunderstood what you meant. You're suggesting that the type for `annotations` should just be `Struct`. That makes perfect sense to me. I think we should change the protobuf definition, and note that it corresponds to the spec, which defines `annotations as a `map[string]Value`. I'll open an issue for this.

Reply:
Ok, I opened #263 to address this.